### PR TITLE
Bitflag returns from component signals

### PR DESCRIPTION
**Pull request by ninjanomnom**
Useful for getting more complex information out of your signals than just a simple true or false that also has side effects.

### CODE REVIEW

File: code/__DEFINES/components.dm [hunk 1]
```diff
@@ -23,6 +28,7 @@
 
 // /atom signals
 #define COMSIG_PARENT_ATTACKBY "atom_attackby"			        //from base of atom/attackby(): (/obj/item, /mob/living, params)
+	#define COMPONENT_NO_AFTERATTACK 2								//Return this in response if you don't want afterattack to be called
```
Comment (praisenarsie) [inline code/__DEFINES/components.dm]: indentation?

Comment (ninjanomnom) [reply code/__DEFINES/components.dm]: intentional

File: code/datums/components/spooky.dm [hunk 2]
```diff
@@ -19,7 +19,7 @@
 	if(ishuman(C))
 		var/mob/living/carbon/human/H = C
 		if(istype(H.dna.species, /datum/species/skeleton))
-			return ..() //undeads are unaffected by the spook-pocalypse.
+			return //undeads are unaffected by the spook-pocalypse.
```
Comment (ninjanomnom) [inline code/datums/components/spooky.dm]: \>`..()` in a proc definition

File: code/datums/components/_component.dm [hunk 3]
```diff
@@ -164,24 +164,24 @@
...
-			return FALSE
+			return NONE
 		var/list/sps = C.signal_procs
 		var/datum/callback/CB = LAZYACCESS(sps, sigtype)
 		if(!CB)
-			return FALSE
-		. = CB.InvokeAsync(arglist(arguments))
-		if(.)
+			return NONE
+		. |= CB.InvokeAsync(arglist(arguments))
```
Comment (Cyberboss) [inline code/datums/components/_component.dm]: just `. = ` here

File: code/datums/components/_component.dm [hunk 4]
```diff
@@ -164,24 +164,24 @@
 /datum/proc/SendSignal(sigtype, ...)
 	var/list/comps = datum_components
 	if(!comps)
-		return FALSE
+		return NONE
 	var/list/arguments = args.Copy()
 	arguments.Cut(1, 2)
 	var/target = comps[/datum/component]
+	. = NONE
```
Comment (Cyberboss) [inline code/datums/components/_component.dm]: this can be moved down to above the for

File: code/datums/components/archaeology.dm [hunk 5]
```diff
@@ -20,27 +20,26 @@
...
-			var/obj/item/pickaxe/P = W
-			digging_speed = P.digspeed
-
-		if (digging_speed && isturf(user.loc))
-			to_chat(user, "<span class='notice'>You start digging...</span>")
-			playsound(parent, 'sound/effects/shovel_dig.ogg', 50, 1)
+		return
+		
+	var/digging_speed
+	if (istype(W, /obj/item/shovel))
```
Comment (Cyberboss) [inline code/datums/components/archaeology.dm]: Oh god this file is space indented, reeee

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/components/archaeology.dm (hunk 5 above)
Comment (ninjanomnom) [reply code/datums/components/archaeology.dm]: no it's not, I just removed some unnecessary indentation here

File: code/__DEFINES/components.dm (hunk 1 above)
Comment (optimumtact) [reply code/__DEFINES/components.dm]: why

Comment (ninjanomnom) [reply code/__DEFINES/components.dm]: It doesn't really matter too much but it feels like the best way given the circumstances to indicate sub-grouping. The other option is to add more newlines and/or comments. This is already a heavily commented file, and newlines would effectively double the length in order to maintain consistency.

It looks a little dumb but it's clear what's happening here and doesn't waste space.

**Comment by Cyberboss** [comment]
For the record, I still really hate ComponentActivated, but it's a necessary evil until everything is components

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**